src/main/java/com/vaadin/training/bugrap/ui/events/CommentAttachmentStartedEvent.java

```diff
@@ -0,0 +1,27 @@
+package com.vaadin.training.bugrap.ui.events;
+
+import java.io.Serializable;
+
+import com.vaadin.training.bugrap.ui.model.CommentUploadModel;
+import com.vaadin.ui.Upload;
+
+public class CommentAttachmentStartedEvent implements Serializable {
+
+	private static final long serialVersionUID = -6963823684567948154L;
+
+	private final Upload upload;
+	private final CommentUploadModel commentUploadModel;
+
+	public CommentAttachmentStartedEvent(final Upload upload, final CommentUploadModel commentUploadModel) {
+		this.upload = upload;
+		this.commentUploadModel = commentUploadModel;
+	}
+
+	public CommentUploadModel getCommentUploadModel() {
```

Comment:
Sharing a model with event might be a bit hacky solution. Other way to do this might be to wrap upload-button into something that is aware of the model that is tied to button. The actual transformation from model-aware button to model-aware display-panel can be done in AddCommentPanel-component by registering another upload started listener to button.